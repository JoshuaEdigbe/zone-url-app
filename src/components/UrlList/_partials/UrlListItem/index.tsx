import React from "react";
import { UrlItem } from "../../../../types";

interface Props {
  itemData: UrlItem;
  onDelete: (itemData: UrlItem) => void;
}

const UrlListItem: React.FC<Props> = ({ itemData, onDelete }) => {
  const handleDelete = () => {
    onDelete(itemData);
  };
  return (
    <li key={itemData?.id}>
      <div className="urls-container">
        <span className="short-url-text">{itemData.shortUrl}</span>
        <span className="old-url-text">{itemData.userUrl}</span>
      </div>
      <button className="delete-btn" onClick={handleDelete}>
        delete
      </button>
    </li>
  );
};

export default UrlListItem;
