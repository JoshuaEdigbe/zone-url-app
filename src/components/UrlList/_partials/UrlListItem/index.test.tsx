import React from "react";
import { render, fireEvent } from "@testing-library/react";
import UrlListItem from ".";

describe("UrlListItem component", () => {
  const mockItemData = {
    shortUrl: "http://short.est/abcd1234",
    userUrl: "http://example.com",
    id: "1",
  };

  it("renders without crashing", () => {
    render(<UrlListItem itemData={mockItemData} onDelete={() => {}} />);
  });

  it("displays short and user URLs correctly", () => {
    const { getByText } = render(
      <UrlListItem itemData={mockItemData} onDelete={() => {}} />
    );

    const shortUrlText = getByText(mockItemData.shortUrl);
    const userUrlText = getByText(mockItemData.userUrl);

    expect(shortUrlText).toBeInTheDocument();
    expect(userUrlText).toBeInTheDocument();
  });

  it("calls onDelete when delete button is clicked", () => {
    const onDeleteMock = jest.fn();
    const { getByText } = render(
      <UrlListItem itemData={mockItemData} onDelete={onDeleteMock} />
    );

    const deleteButton = getByText("delete");
    fireEvent.click(deleteButton);

    expect(onDeleteMock).toHaveBeenCalledWith(mockItemData);
  });

  it("matches snapshot", () => {
    const { container } = render(
      <UrlListItem itemData={mockItemData} onDelete={() => {}} />
    );
    expect(container).toMatchSnapshot();
  });
});
