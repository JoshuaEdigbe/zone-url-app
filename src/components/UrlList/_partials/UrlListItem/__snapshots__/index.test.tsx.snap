// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlListItem component matches snapshot 1`] = `
<div>
  <li>
    <div
      class="urls-container"
    >
      <span
        class="short-url-text"
      >
        http://short.est/abcd1234
      </span>
      <span
        class="old-url-text"
      >
        http://example.com
      </span>
    </div>
    <button
      class="delete-btn"
    >
      delete
    </button>
  </li>
</div>
`;
